plugins/opencost/README.md

```diff
@@ -0,0 +1,79 @@
+# OpenCost
+
+Welcome to the [OpenCost](https://opencost.io) plugin!
+
+Currently this is a port of the [OpenCost UI](https://github.com/opencost/opencost/tree/develop/ui), but we will continue to expand it to expose any relevant data or pre-configured views that may be preferred.
+
+All of the code was originally ported from https://github.com/opencost/opencost/blob/develop/ui/ which is under by the Apache v2 License and also managed by the CNCF.
+
+## Installation
+
+1.  Add the OpenCost dependency to the `packages/app/package.json`:
+    ```sh
+    # From your Backstage root directory
+    yarn add --cwd packages/app @backstage/plugin-opencost
+    ```
+2.  Add the `OpenCostPage` to your `packages/app/src/App.tsx`:
+
+```tsx
+import { OpenCostPage } from '@backstage/plugin-opencost';
+```
+
+and
+
+```tsx
+<FlatRoutes>
+  …
+  <Route path="/opencost" element={<OpenCostPage />} />
+</FlatRoutes>
+```
+
+3.  Add link to OpenCost to your sidebar
+
+    ```typescript
+    // packages/app/src/components/Root/Root.tsx
+     import MoneyIcon from '@material-ui/icons/MonetizationOn';
+
+     ...
+
+     export const Root = ({ children }: PropsWithChildren<{}>) => (
+       <SidebarPage>
+         <Sidebar>
+           ...
+          <SidebarItem icon={MoneyIcon} to="opencost" text="OpenCost" />
+           ...
+         </Sidebar>
+       </SidebarPage>
+     );
+
+    ```
+
+## Plugin Configuration
+
+Since OpenCost doesn't have any authentication at this point, you just need to give API access to the plugin to access your data.
+
+If you haven't set up an ingress rule, you can port-forward the API with
+
+```
+kubectl -n opencost port-forward deployment/opencost 9003
+```
+
+Add the following to your `app-config.yaml`:
+
+```yaml
+opencost:
+  baseUrl: http://localhost:9003
```

Comment:
I wonder if we could integrate `opencost` with the Kubernetes plugin backend and show cost across multiple clusters, you could then re-use the authentication we've built into the k8s plugin

Reply:
Are you describing switching across instances, or federating multiple OpenCosts into a single cluster? Today OpenCost is primarily single-cluster, you can roll up the data to a larger time series database like Mimir or Thanos, but it's not well-documented. We could definitely add the feature if there's interest, but probably not in this first pass.

Reply:
The Kubernetes plugin backend is basically a service within backstage that can fetch information from Kubernetes clusters, so I'm wondering if we could reuse that to source the baseUrl for the cluster we want to look at

Reply:
@mclarke47 could you point me to an example of this so I can understand how it might be implemented in the OpenCost plugin? I'll check it out and see if I can refactor accordingly.

Reply:
Sure, [here](https://github.com/backstage/backstage/blob/master/plugins/kubernetes/src/api/KubernetesBackendClient.ts#L124) is the kubernetes frontend asking the backend for the configured clusters

Reply:
After thinking it over, I'm not sure it would make sense to use the Kubernetes backend. Each Kubernetes cluster should have an OpenCost installed, but if the OpenCost deployments are federated to a single UI (which Kubecost does), you would want to specify which OpenCost deployment to point to most likely. We could implement this separately so if there was only 1 Kubernetes cluster it would expose the OpenCost endpoint automatically, but at some point we will need to be able to specify the URL to use as well.

Reply:
Ok,

Separately I'm also building a plugin for Kubernetes clusters and it might make sense to make this available here?

WDYT?

https://github.com/backstage/backstage/pull/18760

Reply:
I think this could be useful once both plugins are merged, and not just for OpenCost. If there was a means for the Kubernetes plugin to show the cluster and provide a link to the OpenCost plugin with the opencost baseUrl already provided that would be compelling. Multiple clusters could be supported, but then we'd need to figure out a bit of navigation for switching between clusters on the OpenCost side, but we have to solve for that anyway. But the K8s cluster + OpenCost pairing would be useful and I could see other K8s services wanting to use the same pattern.